directives/point_of_interest.py

```diff
@@ -32,50 +40,106 @@ class PointOfInterest(Directive):
     has_content = True
     final_argument_whitespace = True
     option_spec = {
+        'id': directives.unchanged,
         'title': directives.unchanged,
         'previous': directives.unchanged,
         'next': directives.unchanged,
         'hidden': directives.flag,
         'class' : directives.class_option,
+        'height': directives.length_or_percentage_or_unitless,
+        'columns': directives.unchanged,
+        'bgimg':directives.uri,
     }
 
     def run(self):
         self.assert_has_content()
+        env = self.state.document.settings.env
         if len(self.arguments) == 0:
             return [nodes.container()]
-        name = self.arguments[0]
-        title_text = ''
-        if 'title' in  self.options:
+
+        if 'title' in self.options:
+            if 'id' in self.options:
+                raise SphinxError(u'Point of interest options can\'t contain both "title" and "id"; one of them should be provided as an argument instead')
+            name = self.arguments[0]
             title_text = self.options['title']
+        else:
+            name = self.options.get('id', ''.join(random.choice(string.ascii_lowercase + string.digits) for i in range(6)))
```

Comment:
`.get('id') or random_string_generation` would be bit faster. Currently, the random string is always generated. Bit better looking code would be:

```
name = self.options.get('id')
if not name:
  choices = string.ascii_lowercase + string.digits
  name = ''.join(random.choice(choices) for i in range(6))
```

p.s. there is still change that the name is not unique.
p.s.s. `choices` is optimization, which is not done automatically by Python

Reply:
I'll change that and add a check to make sure the id is unique. Not quite sure what you mean with the last one though?

Reply:
Just an information why I wrote the example like that. Real programming language would do the optimisation, but python can't.

Reply:
Ah ok, I see. Changed this and in the process noticed that poi info was not correctly cleared from the environment so added a fix for that as well.